### ISSUE TITLE
Add support for vectors and matrices of integers

### ISSUE THREAD

**Issue by parsa** [STEllAR-GROUP/phylanx#414]
For some reason NumPy matrices with 1 dimension are mapped to a vector.

Consider this code:
```python
@Phylanx()
def f(lhs, rhs):
    return lhs + rhs

a = np.array([[1], [2]])
b = np.array([[1, 2]])
r = f(a, b)
print(r)
```
This code results in creation of a `phylanx::ir::node_data` with its `value_` containing a vector instead of a matrix. Note that this only happens for the Python frontend the equivalent code in PhySL works properly.

Update: Here is a simpler piece of code that demonstrates the issue:
```python
@Phylanx()
def f(v):
    return v

a = np.array([[1], [2]])
r = f(a)
print('expected:', a, a.shape)
print('actual:', r, r.shape)
```
```
expected: [[1]
 [2]] (2, 1)
actual: [1. 2.] (2,)
```

**Comment by hkaiser** [comment]
The problem is caused by a couple of conversion issues in the Python translation layer combined with how `numpy.array` is exposed through pybind11. In short, the code above creates a `np.array([[1], [2]], dtype=Int32)` that (by default) is not convertible to a `node_data<double>` but is converted to a `vector<primitive_argument_type>` filled with `std::int64_t` instead. 

The solution would be to support `node_data<std::int64_t>` (which needs to be implemented in any case).

The matter is complicated by the current behavior of our translation layer that wraps all numpy arrays into a PhySL variable (which needs to be changed as well).

FWIW, here is the Python test that can be used to verify the issue was fixed:
```
import numpy as np
from phylanx.ast import *


@Phylanx()
def f(a):
    return shape(a)


a = np.array([[1], [2]])
assert(f(a) == [2, 1])
```
Also, as a workaround, use `np([[1], [2]], dtype=float)`.

**Comment by hkaiser** [comment]
See https://github.com/STEllAR-GROUP/phylanx/tree/fixing_414

**Comment by parsa** [comment]
The implications of @hkaiser's explanation:
* Currently any NumPy array received by Phylanx is treated as if it's of type double
* Phylanx will receive a copy of each NumPy array that is not of floating points types instead of the original NumPy array being mapped to a Phylanx `node_data`

**Comment by hkaiser** [comment]
This should have bee resolved by #459. Please reopen if the issue persists.
